### PR TITLE
Fix IVOID of all geometrical features

### LINKED ISSUE #71: ADQL geometry language feature identifier version inconsistency
I raised this [on the DAL list in 2019](http://mail.ivoa.net/pipermail/dal/2019-September/008183.html) but didn't get a response, so I'll try it here instead...

In the current ADQL 2.1 draft, geometrical functions are labelled using the following Language Feature identifier:
```
ivo://ivoa.net/std/TAPRegExt#features-adql-geo
```
However in ADQL 2.0, the corresponding functionality was marked by the feature (defined in [TAPRegExt 1.0 Sec 2.3](https://www.ivoa.net/documents/TAPRegExt/20120827/REC-TAPRegExt-1.0.html#langs)):  
```
ivo://ivoa.net/std/TAPRegExt#features-adqlgeo
```
Note the lack of a second hyphen.

This looks like a typo, so **I suggest changing `adql-geo` to `adqlgeo`** in the current text.  If not, there is an inconsistency between ADQL 2.0 and 2.1.  I can prepare a PR on request.

This change could be problematic if there are already services out there using the `adql-geo` form as written in the current draft.  However, I suspect that there are few or none.

RegTAP does not record language features, but if I look for all the services claiming to support ADQL 2.1:
```
select ivoid, detail_xpath, detail_value, access_url
from rr.res_detail
natural join rr.capability
natural join rr.interface
where detail_xpath = '/capability/language/version/@ivo-id'
  and detail_value = 'ivo://ivoa.net/std/ADQL#v2.1'
  and standard_id = 'ivo://ivoa.net/std/tap'
  and intf_type='vs:paramhttp'
```
there are 24. The couple I tried (both DaCHS) use `adqlgeo` not `adql-geo` already (i.e. in accordance with my suggested change, and in violation of the draft as currently written).  I suspect the others are mostly DaCHS instances too, so have the same behaviour.

Since I spotted this in 2019 (STILTS v3.2 and later) taplint has issued an Error if the `adql-geo` form is used, like this:
```
E-CAP-KEYX-1 Unknown standard feature key "ivo://ivoa.net/std/TAPRegExt#features-adql-geo" for language ADQL-2.1
```
i.e. taplint currently agrees with my suggested change rather than what's written in the current ADQL 2.1 draft.
So my guess is that this change is not going to cause problems.

This issue has come up now because ESDC are using the `adql-geo` form in their ADQL 2.1 services, as in the current draft, and have hit that taplint error.

### PR DESCRIPTION
**Pull request by gmantele**
This Pull-Request fixes the IVOID of the language features for all geometrical functions.

```
ivo://ivoa.net/std/tapregext#features-adqlgeo
```

instead of

```
ivo://ivoa.net/std/tapregext#features-adql-geo
```

The IVOID of all other ADQL features starts with `ivo://ivoa.net/std/tapregext#features-adql-`. But for geometrical features it should not. Indeed, on the contrary to the other ADQL features, this one is already used in TAP services implementing a draft version of ADQL-2.1. This IVOID being already in existing in-production and IVOA registered services, it cannot be made consistent with the others ADQL language features ; this way, it avoids breaking existing services.

Fixes #71

### CODE REVIEW

**Comment by gmantele** [comment]
I wait for the #74 to be merged, before merging this one. This way, I will rebase it on top of #74 commits and complete the change log. Then, once reviewed and approved, I'll merge it with the master branch.

File: ADQL.tex
Comment (msdemlei) [inline ADQL.tex]: Ah... no, the reason actually is more serious: features-adqlgeo is already defined in TAPRegExt 1.0.  So, I guess we should just say "it's legacy, and the name was coined before it became clear that we will have a lot more features-adql-* identifiers" or so.

**Review by msdemlei — CHANGES_REQUESTED** [review]
Oh, and will you help me remember to push an updated TAPRegExt StandardsRegExt record with the next PR?  That way, all the identifiers we define here will properly resolve, too.

File: ADQL.tex
Comment (gmantele) [reply ADQL.tex]: > Ah... no, the reason actually is more serious: features-adqlgeo is already defined in TAPRegExt 1.0. So, I guess we should just say "it's legacy, and the name was coined before it became clear that we will have a lot more features-adql-* identifiers" or so.

Ho, I see. Is it better now?

**Comment by gmantele** [comment]
> Oh, and will you help me remember to push an updated TAPRegExt StandardsRegExt record with the next PR? That way, all the identifiers we define here will properly resolve, too.

Sure. Do you need a GitHub issue to remember it?

**Comment by gmantele** [comment]
By the way, @msdemlei, have you noticed that I put all IVOIDs in lower-case, especially the `std/TAPRegExt#` part?

In TAPRegExt-1.0, IVOIDs are written in mixed case (i.e. the `TAPRegExt` part). So, is it still ok to write that fully in lower case in ADQL-2.1?

**Review by msdemlei — DISMISSED** [review]
Yeah, let's go.

[I'd even be ok with a more aggressive text to the effect that we are deliberately breaking the 1.0 pattern now that we have so many ADQL features, but that's a matter of taste -- feel free to merge what you have, but feel equally free to remove references to ADQL 2.1 beta implementations]

File: ADQL.tex
Comment (mbtaylor) [inline ADQL.tex]: I would be inclined to cut back on the historical detail here, since it's not really necessary for understanding the standard.  I'd suggest just writing something like "For historical reasons the IVOID for the geometry features does not have this hyphen" and leaving it at that.  But I leave it to @gmantele to decide whether to follow that suggestion.

**Review by mbtaylor — DISMISSED** [review]
Thank you @gmantele, this looks to me like it resolves this issue.  I made one comment which you can follow up if you like.

File: ADQL.tex
Comment (gmantele) [reply ADQL.tex]: Done. I still want to keep a small explanation of why this difference. So I left the TAPRegExt reference.
Is it better?

**Review by mbtaylor — APPROVED** [review]

**Comment by msdemlei** [comment]
On Thu, Mar 09, 2023 at 12:26:29AM -0800, Grégory Mantelet wrote:
> By the way, @msdemlei, have you noticed that I put all IVOIDs in lower-case, especially the `std/TAPRegExt#` part?
>
> In TAPRegExt-1.0, IVOIDs are written in mixed case (i.e. the
> `TAPRegExt` part). So, is it still ok to write that fully in lower
> case in ADQL-2.1?

Yeah, it's much better this way.  Ivoids regrettably are
case-insensitive, and giving them in mixed case then is inviting all
kinds of grief.  All-lowercase FTW.

**Comment by msdemlei** [comment]
On Thu, Mar 09, 2023 at 12:22:18AM -0800, Grégory Mantelet wrote:
> > Oh, and will you help me remember to push an updated TAPRegExt StandardsRegExt record with the next PR? That way, all the identifiers we define here will properly resolve, too.
>
> Sure. Do you need a GitHub issue to remember it?

I don't think so -- if you follow the submission checklist while
preparing the PR
<https://ivoa.net/documents/Notes/IVOATexDoc/20220525/NOTE-ivoatexDoc-1.3-20220525.html#tth_sEc3.12>,
you'll to the point

  If uploading a REC or EN, update (or create) the standards record
  and send it to the RofR

-- and at that point just poke me, because (somewhat stupidly) the
registry keys for ADQL are in TAPRegExt...

**Pull request merged**
